components/centraldashboard/public/components/main-page.js

```diff
@@ -173,10 +173,12 @@ export class MainPage extends utilitiesMixin(PolymerElement) {
      * @param {Event} ev AJAX-response
      */
     _onHasWorkgroupResponse(ev) {
-        const {user, hasWorkgroup, hasAuth} = ev.detail.response;
+        const {user, hasAuth, namespaceCreation} = ev.detail.response;
         this._setIsolationMode(hasAuth ? 'multi-user' : 'single-user');
-        if (hasAuth && !hasWorkgroup) {
-            this.user = user;
+        this.user = user;
+        if (!namespaceCreation) {
+            this._setRegistrationFlow(false);
+        } else {
             this._setRegistrationFlow(true);
```

Comment:
This logic is incorrect you need to make sure that **both** `!hasWorkgroup && namespaceCreation` (when `namespaceCreation` is actually allowed) for the flow to trigger!

Reply:
Should be good now.